### PR TITLE
Refactor SpellCache

### PR DESCRIPTION
**Pull request by jgersti**
see #145

### CODE REVIEW

File: WeakAurasOptions/Cache.lua [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      elseif(IsSpellKnown(spellId)) then
+        bestMatch = spellId;
+      end
+    end
+  end
+
+  return bestMatch and icons[bestMatch];
+end
+
+function WeakAuras.EmplaceIconIntoSpellCache(name, id, icon)
```
Comment (InfusOnWoW) [inline WeakAurasOptions/Cache.lua]: AddIcon

File: WeakAurasOptions/Cache.lua [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      spellCache[name][id] = icon
+    end
+  end
+end
+
+function WeakAuras.GetSpellCache()
+  return spellCache
+end
+
+function WeakAuras.LoadSpellCache(cache, override)
```
Comment (InfusOnWoW) [inline WeakAurasOptions/Cache.lua]: where's the call to this function with override ~= nil?

File: WeakAurasOptions/WeakAurasOptions.lua [hunk 3]
```diff
@@ -43,9 +43,6 @@ end
 local dynFrame = WeakAuras.dynFrame;
 WeakAuras.transmitCache = {};
 
-local spellCache = {};
-local talentCache = {};
```
Comment (InfusOnWoW) [inline WeakAurasOptions/WeakAurasOptions.lua]: The removal of talentcache is unrelated, but with me.

File: WeakAurasOptions/WeakAurasOptions.lua [hunk 4]
```diff
@@ -7128,7 +7061,7 @@ function WeakAuras.CreateFrame()
     local usedIcons = {};
     local num = 0;
     if(subname ~= "") then
-      for name, icons in pairs(spellCache) do
+      for name, icons in pairs(WeakAuras.GetSpellCache()) do
```
Comment (InfusOnWoW) [inline WeakAurasOptions/WeakAurasOptions.lua]: So, in the spirit of making the code oragnization better. I think the whole object should be WeakAuras.spellcache, with methods on it. e.g. WeakAuras.spellcache.Add(), WeakAuras.spellcache.Load(), WeakAuras.spellcache.Get().

And I think this whole matching code could then be e.g. WeakAuras.spellcache.bestMatch( key ).

File: WeakAurasOptions/Cache.lua [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+function spellCache.Get()
+  if cache then
+    return cache
+  else
+    error("spellCache has not been loaded. Call WeakAuras.spellCache.Load(...) first.")
+  end
+end
+
+function spellCache.Load(data, override)
```
Comment (InfusOnWoW) [inline WeakAurasOptions/Cache.lua]: Can you remove the override parameter?

**Comment by Stanzilla** [comment]
This looks ready to me. 

@InfusOnWoW any issues left from your side?
@jgersti the branch now has conflicts which need to be fixed but otherwise good job 👍

**Comment by InfusOnWoW** [comment]
I'll check today. I need to change my setup here a bit to be able to easily test pull requests.

**Comment by InfusOnWoW** [comment]
Looks good to me.

**Pull request merged**